daemon/cmd/ipam.go

```diff
@@ -193,15 +193,34 @@ func (d *Daemon) allocateDatapathIPs(family types.NodeAddressingFamily, fromK8s,
 		}
 	}
 
-	// Coalescing multiple CIDRs. GH #18868
-	if option.Config.EnableIPv4Masquerade &&
-		option.Config.IPAM == ipamOption.IPAMENI &&
-		result != nil &&
-		len(result.CIDRs) > 0 {
-		result.CIDRs, err = coalesceCIDRs(result.CIDRs)
-		if err != nil {
-			return nil, fmt.Errorf("failed to coalesce CIDRs: %w", err)
+	ipfamily := ipam.DeriveFamily(family.PrimaryExternal())
+	var masq bool
+	switch ipfamily {
+	case ipam.IPv4:
+		// Coalescing multiple CIDRs. GH #18868
+		if option.Config.EnableIPv4Masquerade &&
```

Comment:
nit (sorry I didn't catch this earlier, feel free to ignore): You could probably reduce the amount of duplicated code by moving this if statement below the switch and have it read the `masq` variable.